### PR TITLE
Extract kvstore into separate crate

### LINKED ISSUE #3310: Move KvStore out of core/ and into its own crate
Self explanatory

### PR DESCRIPTION
**Pull request by MarkJr94**
#### Problem
The store is largely self-contained and doesn't belong in the core crate.
#### Summary of Changes
Extract `core/src/kvstore` into a new crate

Fixes #3310

### CODE REVIEW

**Comment by mvines** [comment]
Please add kvstore to the list of crates here:
https://github.com/solana-labs/solana/blob/master/ci/publish-crate.sh#L14
Be sure to list it ahead of any crates that depend on it.

File: kvstore/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+[package]
+name = "solana-kvstore"
+description = "Embedded Key-Value store for solana"
+version = "0.0.1"
+homepage = "https://solana.com/"
+readme = "../README.md"
```
Comment (mvines) [inline kvstore/Cargo.toml]: IMO remove this readme entry, we only need this in one crate (core)

File: Cargo.toml [hunk 2]
```diff
@@ -89,5 +90,5 @@ members = [
     "upload-perf",
     "vote-signer",
     "wallet",
-]
-exclude = ["programs/bpf/rust/noop"]
+    ]
```
Comment (mvines) [inline Cargo.toml]: remove stray whitespace changes please

File: kvstore/Cargo.toml (hunk 1 above)
Comment (MarkJr94) [reply kvstore/Cargo.toml]: done

File: Cargo.toml (hunk 2 above)
Comment (MarkJr94) [reply Cargo.toml]: done

**Pull request merged**